### PR TITLE
knex: reference correct Promise implementation

### PR DESCRIPTION
**Pull request by jeffkenney**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [x] Run `npm run lint package-name` if a `tslint.json` is present.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/tgriesser/knex/blob/master/src/client.js#L1
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

### CODE REVIEW

File: knex/index.d.ts [hunk 1]
```diff
@@ -7,6 +7,7 @@
 /// <reference types="node" />
 
 import * as events from "events";
+import * as Promise from "bluebird";
```
Comment (DanielRosenwasser) [inline knex/index.d.ts]: Can you switch these to

```ts
import events = require("events");
import Promise = require("bluebird");
```

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by jeffkenney** [comment]
Hey @DanielRosenwasser ,

I made the requested changes. As a quick aside, why is `import Promise = require('bluebird');` preferred over `import * as Promise from 'bluebird';`?

**Comment by DanielRosenwasser** [comment]
Technically per the ES spec, a namespace import is not constructable, callable, etc. Using the `import x = require('foo')` syntax accurately models that this is a CommonJS import.

**Review by DanielRosenwasser — APPROVED** [review]

File: knex/knex-tests.ts [hunk 2]
```diff
@@ -1,5 +1,6 @@
 "use strict";
 import * as Knex from 'knex';
+import * as Promise from 'bluebird';
```
Comment (DanielRosenwasser) [inline knex/knex-tests.ts]: Sorry about this - can you change this to `import x = require("foo")` here as well please?

**Review by DanielRosenwasser — CHANGES_REQUESTED** [review]

**Comment by jeffkenney** [comment]
@DanielRosenwasser,
Tests file updated; thanks for the explanation.

Are the following methods then the best practices for imports?

```typescript
// from commonjs modules using `module.exports`
import x = require('y');
```

```typescript
// from es6 modules using `export default`
import x from 'y';
```

```typescript
// from es6 modules with named exports
import * as x from 'y';
x.a();
// or
import { a } from 'y';
a();
```

**Comment by DanielRosenwasser** [comment]
@jeffkenney yes, I'd say that's accurate. Thanks for the changes!

**Review by DanielRosenwasser — APPROVED** [review]

**Pull request merged**